src/main/java/com/github/osisoft/ocs_sample_library_preview/sds/StreamsClient.java

```diff
@@ -1104,6 +1105,61 @@ public String getRangeValuesInterpolated(String tenantId, String namespaceId, St
         return response;
     }
 
+    /**
+     * gets values at the indexes provided
+     * 
+     * @param tenantId    tenant to work against
+     * @param namespaceId namespace to work against
+     * @param streamId    the stream to get values from
+     * @param index       the list of indexes to retrieve values for
+     * @return string of the array of values
+     * @throws SdsError any error that occurs
+     */
+    public String getIndexCollectionValues(String tenantId, String namespaceId, String streamId,
+            ArrayList<String> index) throws SdsError {
+        URL url = null;
+        HttpURLConnection urlConnection = null;
+        String response = "";
+
+        try {
+            String indexString = "";
+            for (int i = 0; i < index.size(); i++) {
```

Comment:
also, is the size function of an ArrayList trivial? I can't find this online, but if the arraylist needs to do any computation to determine its size, it's more efficient to get it and store it in a temp variable, then do the comparison against that each time. The size won't change over the course of the loop, so computing the size once and using that answer every time is safe.

Reply:
looks like java makes a new contiguous block of memory for each resize attempt, as opposed to using something like a linked list under the hood to make it infinitely expandable, so the size function should be trivial

Reply:
Yeah, the size is stored in the ArrayList, meaning that it is not calculated every call.